kits/daisy/src/input/input_manager_mobile_io.cpp

```diff
@@ -0,0 +1,137 @@
+#include "input_manager_mobile_io.hpp"
+#include "lookup.hpp"
+#include "xml_util/pugixml.hpp"
+#include "group_feedback.hpp"
+
+#include <iostream> // Note: for debugging output.
+
+namespace hebi {
+namespace input {
+
+InputManagerMobileIO::InputManagerMobileIO()
+{
+  if (!reset())
+    std::cout << "Could not find joystick input board!" << std::endl;
+}
+
+bool InputManagerMobileIO::reset()
+{
+  // Cleanup if we already have one!
+  if (group_)
+    group_.reset();
+
+  // Look for joystick (IO board)
+  hebi::Lookup lookup;
+  long timeout_ms = 2000; // use a 2 second timeout
+  group_ = lookup.getGroupFromNames({"HEBI"}, {"Mobile IO"}, timeout_ms);
+  if (!group_)
+    return false;
+
+  group_->addFeedbackHandler([this] (const GroupFeedback& fbk)
+  {
+    // TODO: use 'std::atomic' variables for state here? Add mutex?  Does this
+    // slow things down due to lock contention?
+    const auto& analog = fbk[0].io().a();
+    const auto& digital = fbk[0].io().b();
+    if (analog.hasFloat(1) && analog.hasFloat(2) && analog.hasFloat(3) &&
+        digital.hasInt(1))
+    {
+      left_horz_raw_ = analog.getFloat(1);
+      left_vert_raw_ = analog.getFloat(2);
+
+      slider_1_raw_ = analog.getFloat(3);
+
+      right_horz_raw_ = analog.getFloat(7);
+      right_vert_raw_ = analog.getFloat(8);
+
+      // Note: only care about edge triggers down here
+      bool new_mode_button_state = (digital.getInt(7) == 1);
+      if (new_mode_button_state && !prev_mode_button_state_)
+        ++num_mode_toggles_;
+      prev_mode_button_state_ = new_mode_button_state;
+
+      // Check for "Quit"
+      if (digital.getInt(8) == 1) {
+        has_quit_been_pushed_ = true;
+      }
+    }
+  });
+  group_->setFeedbackFrequencyHz(1000); // Don't want to miss button presses!
```

Comment:
I don't think turning the frequency up all of the way will result in a better chance of getting button presses, and risks just dropping a ton more packets because wifi isn't going to handle this well whatsoever. This is also problematic at the user facing side, since it portrays the assumption that it's going to work well like this

Reply:
Previously, this hadn't been going over wifi -- I hadn't thought about this since that change was made.

In this case, wouldn't wifi just buffer the packets instead of dropping them?  I feel like this would still make it more likely to catch a momentary button press that happened for a fraction of a second, although I agree that this isn't as ideal as when wired.

Overall, for this it would be nice to just use a "toggle button" instead of a "momentary" switch on the I/O app -- I'll add an issue to address this.

Reply:
UDP would potentially buffer the packets, but that is entirely implementation specific. Phones probably don't even acknowledge touch events at 1000Hz (it would only make sense for them to ack them at the 60Hz the screen is locked rendering at)

Reply:
Will fix ASAP